### ISSUE TITLE
Dataverse API

### ISSUE THREAD

**Issue by TomMiksa** [samihabakri/D2S-Dataverse#18]
Create a website with a simple formular that allows to create a dataverse within Dataverse.

Your website must use Dataverse API to make a call to create a new dataverse. 

(all you need to do is to call the REST API of dataverse)

**Comment by AhmadHirthani** [comment]
Create a Dataverse
Generates a new dataverse under $id. Expects a JSON content describing the dataverse, as in the example below. If $id is omitted, a root dataverse is created. $id can either be a dataverse id (long) or a dataverse alias (more robust).

POST http://$SERVER/api/dataverses/$id?key=$apiKey

Download the JSON example file and modified to create dataverses to suit your needs. The fields name, alias, and dataverseContacts are required. The controlled vocabulary for dataverseType is

DEPARTMENT
JOURNALS
LABORATORY
ORGANIZATIONS_INSTITUTIONS
RESEARCHERS
RESEARCH_GROUP
RESEARCH_PROJECTS
TEACHING_COURSES
UNCATEGORIZED

ref: http://guides.dataverse.org/en/latest/api/native-api.html

**Comment by muisawe** [comment]
we are going to use demo.dataverse.org API so is there is any problem of using it ?!

**Comment by TomMiksa** [comment]
I was originally assuming that you will use one of your instances. If you the demo server allows you to make API calls, then you can go with it as well.

**Comment by muisawe** [comment]
Yeah they are mentioned in the guide that we can use it for testing

**Comment by muisawe** [comment]
there is some problem on dataverse API, actually I created an issue  on  [IQSS/dataverse](https://github.com/IQSS/dataverse/issues/5865) 
and there is a lot of issues that same as mine. 
you can test the API yourself using [API Tester](https://apitester.com/shared/checks/0f86f5afcbbd4f20912d2dc32a13d69b)
it's Ready to try just you can open the URL then click on the test button. 

to be honest I wrote PHP code  It's ready to work but the problem is on the API itself.

**Comment by TomMiksa** [comment]
Please try the PyAPI as mentioned here: https://github.com/IQSS/dataverse/issues/5859#issuecomment-494740647

**Comment by muisawe** [comment]
there is an error  occurred while I was trying to test pyDataverse package. 
I created an [issue](https://github.com/AUSSDA/pyDataverse/issues/7) on its repository 2 days ago and nobody has responded yet, so I'm still waiting

**Comment by muisawe** [comment]
I tried to use PyDataverse but when I was trying to create a dataverse, it's created but doesn't appear.

![image](https://user-images.githubusercontent.com/26344769/58574963-7c376100-8241-11e9-92b3-2b4338042654.png)

**Comment by TomMiksa** [comment]
Try harder... check with a local version of Dataverse and check what logs say... 
Try different API calls, etc...
